[Lee Zheng Han] iP

solarrabbit99 opened the pull request:
# Zheng Han's Markdown
## Lists
### Unordered Lists
- This is an un-ordered list
- Second element of un-ordered list
### Ordered Lists
1. First element
2. Second
## Code block
```java
public static void main(String[] args) {
  System.out.println("Follow me on instagram at zhenghanlee");
}
```
## Task list
- [x] CS2103T
- [ ] CS3230
- [ ] CS2101
- [ ] GEQ1000
- [ ] ST2334

## Others
> **Click** [here](https://www.google.com) _for_ `google`, that's ~~all~~ everything. 😉

Commit:
Implemented bot echoing and termination

Commit:
Implemented list for adding and printing elements

Commit:
Implemented marking tasks as completed

Commit:
Implemented support for todo, deadline, and event tasks

Commit:
Added tests to runtest.sh

Commit:
Add error handling for empty command arguments and invalid commands

Commit:
Task is now abstract

Commit:
Update tests for error handling

Commit:
Added DukeException class

Commit:
Include the use of DukeExceptions subclasses for invalid commands/arguments

Commit:
Implement delete command and improve styling

Commit:
Added comment to explain the use of command registry

Commit:
Updated testcases for removal of tasks

Commit:
Create basic structure for SQLite database

Commit:
Add remove task logic for SQLite database

Commit:
Add mark task completed logic for SQLite database

Commit:
Fix SQLite Library missing and create counter testcases to keep database clean for runtest

Commit:
Implement Task deadlines/dates using LocalDate

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Integrate the use of duke exceptions to handle database access and io exceptions

Commit:
Implement dependency injection on commands requiring dukelist to operate on

Commit:
Change dependecy injected to dukelist instead of duke to remove need of list getter

Commit:
Create UserInterface class for managing responses

Commit:
Move regex logic for date to command instead of task constructor

Commit:
Update test cases for localdate

Commit:
Assign reading input to UserInterface

Commit:
Convert project into maven project and add test classes

Commit:
Include maven sade plugin to shade sqlite jdbc driver into jar

Commit:
Exclude junit classes in shade plugin configurations

Commit:
Add more javadoc to public classes and methods

Commit:
Improve code standard to adhere to coding standards

Commit:
Implement finding of tasks by keyword/phrase

Commit:
Fix find command not registered and sql exception on finding tasks

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Add javadoc for find task method in user interface

Commit:
Fix javadoc for ListCommand class

Commit:
Correct getSqlConnection naming to naming convention

Commit:
Initialize gradle build tool

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Add checkstyle and suppressions xml files

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Rename SQLite.java to sql.java

Commit:
rename Sql.java back to Sqlite.java

Commit:
Remove redundant text-ui-testing and target folder

Commit:
Remove trailing whitespaces in Database.java

Commit:
Add javafx dependency

Commit:
Add launcher and convert duke to javafx application

Commit:
Fix checkstyle errors

Commit:
Finish simple duke gui layout

Commit:
Implement duke commands on gui

Commit:
Merge branch 'branch-Level-10'

Tianqi-Zhu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,157 @@
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+
+        userInput.setPrefWidth(325.0);
+
+        sendButton.setPrefWidth(55.0);
+
+        AnchorPane.setTopAnchor(scrollPane, 1.0);
+
+        AnchorPane.setBottomAnchor(sendButton, 1.0);
+        AnchorPane.setRightAnchor(sendButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput, 1.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+
+        sendButton.setOnMouseClicked((event) -> {
+            dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
+            userInput.clear();
+        });
+
+        userInput.setOnAction((event) -> {
+            dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
+            userInput.clear();
+        });
+
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+
+        sendButton.setOnMouseClicked((event) -> {
+            handleUserInput();
+        });
+
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
+
+        ui.greet();
+    }
```
This method looks super long. Maybe it is better to split it into several shorter methods.

solarrabbit99 replied:
Indeed XD

chingh20 commented on src/main/java/duke/DukeList.java:
```diff
@@ -0,0 +1,95 @@
+     *
+     * @return message on the current list size
+     */
+    public String currentSizeMessage() {
```
You might want to change the method name so that it starts with a verb. Perhaps getCurrentSizeMessage?

solarrabbit99 replied:
Yes, that is better! Thanks for the suggestion!

Tianqi-Zhu commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,45 @@
+     * provide its response to the given ui.
+     *
+     * @param dukeList list of tasks from which the task will marked as completed
+     * @param ui       user interface for displaying responses
```
It seems that there is an unnecessary space here.

solarrabbit99 replied:
Thanks for the review! It is a linter that auto-formats the param description to auto-align itself. I thought it was ok to leave it as such, but thanks anyways!

chingh20 commented on src/main/java/duke/database/Database.java:
```diff
@@ -0,0 +1,188 @@
+     * @param name        name of task
+     * @param isCompleted {@code true} if the task has been completed
+     * @param date        date of the task, nullable
+     * @return
```
I think you might have forgotten to finish writing the comment for @return

solarrabbit99 replied:
Haha, indeed...

chingh20 commented on src/main/java/duke/command/CommandManager.java:
```diff
@@ -0,0 +1,60 @@
+     * Executes the default behaviour with the input if there are no commands
+     * called.
+     */
+    private void defaultExec() {
```
maybe rename the method as executeDefult?

solarrabbit99 replied:
Yep, that's better!

chingh20 commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,35 @@
+     * Creates a command that lists tasks in the given dukelist onto the specified
+     * ui.
+     *
+     * @param dukeList of which tasks
```
might be better if you write @param dukeList User's todo list (Start the description with a capitalized letter). Remember to put a punctuation after each parameter description :)

solarrabbit99 replied:
It seems like capitalising the first letter of the description isn't a common practice in general, if I were to direct you the java's String class's [documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html). Again, I may be wrong and that it is part of the module's guideline. Thanks anyways!

solarrabbit99 replied:
On the point of having punctuation, I'm also not sure if it is a [convention](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html) or a standard in the module. It does make sense that it looks neater that way, however.

chingh20 reviewed (commented):
Good job on following the coding standard! I couldn't find many violations to comment on. One small thing I guess you can pay more attention to is the parameter description for comments. Make sure to end each description with a punctuation. It's not necessary to align one parameter description with others, and starting the description with a capitalize letter might also make it neater.  Overall, your code was very nice and easy to read~ well done :))

Commit:
Add assertions to Command classes

DeleteCommand, DoneCommand and FindCommand classes execute a database
update/query that returns a Task or List of Tasks. The exec methods
do checks on invalid arguments.

We are not entirely certain that the database updates/queries will
somehow return null by mistake.

Assertions are added to further ensure that the database updates/
queries returns a non-null value.

Assertions can prematurely terminate the process before it cascades
to situations/problems that are hard to reverse or fix,

Commit:
Improve code quality for UserInterface and Database classes

solarrabbit99 committed:
Merge pull request #1 from zhenghanlee/branch-A-Assertions

Add assertions to Command classes

solarrabbit99 committed:
Merge pull request #2 from zhenghanlee/branch-A-CodeQuality

Improve code quality for UserInterface and Database classes

Commit:
Add fixed duration task type DurationTask

Commit:
Merge branch 'branch-B-FixedDurationTasks'

Commit:
Add getScrollPane() method for Duke class

start() method seems rather long and difficult to read.

Let's abstract out the configurations on the scroll pane as a method
by itself.

By adding getScrollPane() method and making it private, there will
be a clearer separtion of concerns.

Commit:
Downgrade build java version to 11

Commit:
Add third-party libraries acknowledgements in README

solarrabbit99 committed:
Set theme jekyll-theme-architect

Commit:
Add Ui.png to docs folder

Commit:
Update User Guide

Commit:
Change duke and user profile picture

Commit:
Fix handleUserInput method javadoc for Duke class
